hack/tools/release/notes.go

```diff
@@ -104,6 +106,64 @@ func increaseDateByOneDay(date string) (string, error) {
 	return datetime.Format(layout), nil
 }
 
+const (
+	missingAreaLabelPrefix   = "MISSING_AREA"
+	areaLabelPrefix          = "area/"
+	multipleAreaLabelsPrefix = "MULTIPLE_AREAS["
+)
+
+type githubPullRequest struct {
+	Labels []githubLabel `json:"labels"`
+}
+
+type githubLabel struct {
+	Name string `json:"name"`
+}
+
+func getAreaLabel(merge string) (string, error) {
+	// Get pr id from merge commit
+	prID := strings.Replace(strings.TrimSpace(strings.Split(merge, " ")[3]), "#", "", -1)
+
+	cmd := exec.Command("gh", "api", "repos/kubernetes-sigs/cluster-api/pulls/"+prID) //nolint:gosec
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+
+	pr := &githubPullRequest{}
+	if err := json.Unmarshal(out, pr); err != nil {
+		return "", err
+	}
+
+	var areaLabels []string
+	for _, label := range pr.Labels {
+		if area, ok := trimAreaLabel(label.Name); ok {
+			areaLabels = append(areaLabels, area)
+		}
+	}
+
```

Comment:
What about using the github client instead of shelling out to the `gh` cli? Something like this should work, and fail fast if there's no github token as this is very likely to hit a rate limit.

```suggestion
	prID, err := strconv.Atoi(strings.Replace(strings.TrimSpace(strings.Split(merge, " ")[3]), "#", "", -1))
	if err != nil {
		return "", err
	}
	var client *github.Client
	token := os.Getenv("GITHUB_TOKEN")
	if len(token) == 0 {
		return "", errors.New("A GITHUB_TOKEN is required as an environment variable to avoid rate limiting")
	}
	ts := oauth2.StaticTokenSource(
		&oauth2.Token{AccessToken: token},
	)
	tc := oauth2.NewClient(context.TODO(), ts)
	client = github.NewClient(tc)

	pr, resp, err := client.PullRequests.Get(context.TODO(), "kubernetes-sigs", "cluster-api", prID)
	if err != nil {
		fmt.Println(resp)
		return "", err
	}
	var areaLabels []string
	for _, label := range pr.Labels {
		if area, ok := trimAreaLabel(*label.Name); ok {
			areaLabels = append(areaLabels, area)
		}
	}


```

Reply:
I was thinking that using the `gh` cli simplifies the auth problem. `GITHUB_TOKEN` is just one of the ways to authenticate, and `gh` manages all of them. It also allows you to authenticate and generate tokens automatically without having to go manually to the GitHub UI and export them to your terminal session.

If you feel strongly, I can change it. But if the only concern is rate limiting, I think using `gh` simplifies the experience.

Btw, `gh` won't let you run this API request without being authenticated:
```
To get started with GitHub CLI, please run:  gh auth login
Alternatively, populate the GH_TOKEN environment variable with a GitHub API authentication token.
```

I haven't run into rate limiting using `gh`, even with 20 concurrent requests.

Reply:
Makes sense - my only question is whether this is a barrier to entry for new teammember - needing the program installed etc.

But I'm fine with the exec - which this tool does for `git` too. WDYT about having a check in main to see if the required external programs are installed and output a useful error if not?

Reply:
Ah that's a good idea, added.